thread_pool.hpp

```diff
@@ -144,18 +144,22 @@ class thread_pool
             block_size = 1;
             num_blocks = (ui32)total_size > 1 ? (ui32)total_size : 1;
         }
+        num_blocks--;
         std::atomic<ui32> blocks_running = 0;
         for (ui32 t = 0; t < num_blocks; t++)
         {
             T start = ((T)(t * block_size) + the_first_index);
-            T end = (t == num_blocks - 1) ? last_index + 1 : ((T)((t + 1) * block_size) + the_first_index);
+            T end = ((T)((t + 1) * block_size) + the_first_index);
             blocks_running++;
```

Comment:
This line would not be needed if `blocks_running` were initialized to `num_blocks`.